iceoryx_dust/include/iceoryx_dust/internal/cxx/convert.inl

```diff
@@ -240,6 +240,19 @@ inline bool convert::fromString<unsigned long>(const char* v, unsigned long& des
 }
 #endif
 
+#if defined(__GNUC__) && (INTPTR_MAX == INT32_MAX)
+/// introduced for 32-bit arm-none-eabi-gcc since uintptr_t is not uint32_t despite it has the same size
+/// who knows why ¯\_(ツ)_/¯
```

Comment:
:laughing:

Reply:
Yeah, this took a fair bit head-scratching ...and hating C++

Reply:
Therefore we should rewrite everything in Rust :laughing: